Abstract revisions (#42)
greenelab/xswap-manuscript: content/01.abstract.md

## Abstract {.page_break_before}

Important tasks in biomedical discovery such as predicting gene functions, gene-disease associations, and drug repurposing opportunities are often framed as network edge prediction.
The number of edges connecting to a node, termed degree, can vary greatly across nodes in real biomedical networks.
If degree strongly influences edge prediction, then imbalance or bias in the distribution of degrees could lead to nonspecific or misleading predictions.
We introduce a network permutation framework to quantify the effects of node degree on edge prediction.
Our framwork decomposes performance into the proportion attributable to degree and the proportion attributable to edge-dependent inference.

> Additionally, our permutation-derived edge prior quantifies the probability of a connection based only on node degree.

Reviewer: ```suggestion
Our permutation-derived edge prior quantifies the probability of a connection based only on node degree.
```

Reviewer: I think you could save a word on "additionally"

> This feature shows excellent discrimination and calibration for 20 biomedical networks (16 bipartite, 3 undirected, 1 directed), with AUROCs frequently exceeding 0.85.

Reviewer: ```suggestion
The edge prior shows excellent discrimination and calibration for 20 biomedical networks (16 bipartite, 3 undirected, 1 directed), with AUROCs frequently exceeding 0.85.
```

Reviewer: agree w/ @dhimmel

> Feature performance exceeding the edge prior is attributable to factors other than degree and is often only a small portion of overall performance.

Reviewer: ```suggestion
Feature performance exceeding the edge prior is attributable to factors other than degree, and we find that is often only a small portion of overall performance.
```

Reviewer: I tried to emphasize this a bit more. I agree with @dhimmel that this is the most important point.

> In a comparison involving permutated networks and the permutation-derived edge prior, we demonstrate that degree's predictive performance diminishes when the networks used for training and testing were generated using distinct methods and hence have large differences in degree distribution.

Reviewer: ```suggestion
Degree's predictive performance diminishes when the networks used for training and testing, despite measuring the same biological relationships, were generated using distinct techniques and hence have large differences in degree distribution.
```

Reviewer: Tried to emphasize that the networks aim to measure the same thing in different ways.

> Accordingly, we suggest using the edge prior as a baseline for edge prediction and degree-preserving permutation to investigate a feature's ability to capture degree and non-degree information.

Reviewer: ```suggestion
Researchers seeking to predict new or missing edges in biological networks should use the edge prior as a baseline.
Degree preserving permutation allows researchers to directly investigate which features used for prediction capture degree and which also capture additional information.
```

Reviewer: I tried to emphasize that there are two different ways to benefit from this here.

## Abstract {.page_break_before}

Important tasks in biomedical discovery, such as repurposing drugs or predicting the function of genes and their association to disease, are often framed as network edge prediction problems.

> In most real-world networks, the number of edges per node (called "degree") varies drastically.

Reviewer: ```suggestion
The number of edges that each node has, termed degree, varies drastically across nodes in real world networks.
```

> Since degree strongly influences edge prediction, any imbalance or bias in the distribution of degrees leads to nonspecific or misleading predictions.

Reviewer: You haven't made the case that degree strongly influences edge prediction yet. It hasn't come up. You could say _If_...

We introduce a framework to quantify the effect of node degree on edge prediction.

> Using degree-preserving network permutations, we define an "edge prior" to quantify the probability that two nodes are connected based only on their degree.

Reviewer: Is there a reason for edge prior to be in scare quotes?

Applying the edge prior to 20 different biomedical networks (16 bipartite, 3 undirected, 1 directed),

> we demonstrate that node degree information alone is capable of remarkable performance, with AUROCs frequently exceeding 85%.

Reviewer: Doesn't need a newline. Not sure I would use the term "remarkable" here. It can be dangerous to tell your readers what is remarkable, as they may find it expected.

> The predictive performance of degree diminishes on testing networks that were generated using distinct methods and hence show little degree correlation with the training network.

Reviewer: Does this mean that thinking of degree (and consequently xswap) is not actually valuable? I know what you are going for here, but I don't know if most readers will.

Author: My position on this is that the edge prior is important to assess whether degree generalizes to the test set. For most test sets, it probably will. I think the point is that for some tests sets / applications, degree will generalize less. I would be OKAY removing this from the abstract, at least until we have a more rigorous recommendation on how to approach such a situation. Given the word count, I think this is potentially the least essential point.

Accordingly, we suggest using the edge prior as baseline for edge prediction tasks.
Performance exceeding the baseline is attributable to factors other than degree alone and is often only a small portion of overall performance.

> We provide an open source Python package, named [`xswap`](https://github.com/hetio/xswap) after the permutation algorithm, that allows users to permute their own networks and compute edge priors.

Reviewer: I don't know of any authoring system other than manubot that will let you include links in the abstract. I would suggest removing this or making the URL a parenthetical.

## Abstract {.page_break_before}

Important tasks in biomedical discovery such as predicting gene functions, gene-disease associations, and drug repurposing opportunities are often framed as network edge prediction.

> The number of edges connecting to a node, termed degree, can vary greatly across nodes in real biomedical networks.

Author: does "across nodes" really get across what we mean (that within the same network degree varies)? It's a bit redundant IMO because degree implies  "across nodes"

Reviewer: I thought this was trying to also introduce differences degree distribution, not just about the individual node degree. If the idea is just to introduce degree, then I think you need a follow-up section somewhere that also notes that the distribution can vary. The next sentence does some of this lifting, but it just says that the distribution can be biased - not that it is also different across networks.

If degree strongly influences edge prediction, then imbalance or bias in the distribution of degrees could lead to nonspecific or misleading predictions.
We introduce a network permutation framework to quantify the effects of node degree on edge prediction.

> Our framwork decomposes performance into the proportion attributable to degree and the proportion attributable to edge-dependent inference.

Author: framwork typo

Not sure "edge-dependent inference" is the best term, but we will keep brainstorming

> Additionally, our permutation-derived edge prior quantifies the probability of a connection based only on node degree.

Author: "probability of a connection" is clearer as "probability that an edge exists"... or not?

> This feature shows excellent discrimination and calibration for 20 biomedical networks (16 bipartite, 3 undirected, 1 directed), with AUROCs frequently exceeding 0.85.

Author: "This feature" would be clearer as "The edge prior"? I think clarity is more important than avoiding repetition in this case.

> Feature performance exceeding the edge prior is attributable to factors other than degree and is often only a small portion of overall performance.

Author: > is often only a small portion of overall performance.

IMO this is the most important point of the manuscript along with we have the edge prior to assess this. Abstract is stronger on this point now, but still it would be easy for a reader to not get this as a main takeaway. Probably OKAY for now. We can always touch up later.

> In a comparison involving permutated networks and the permutation-derived edge prior, we demonstrate that degree's predictive performance diminishes when the networks used for training and testing were generated using distinct methods and hence have large differences in degree distribution.

Author: > In a comparison involving permutated networks and the permutation-derived edge prior

This seems unessential.

Accordingly, we suggest using the edge prior as a baseline for edge prediction and degree-preserving permutation to investigate a feature's ability to capture degree and non-degree information.

> We released our methods as an open-source Python package on GitHub (https://github.com/hetio/xswap/) and the Python Package Index (https://pypi.org/project/xswap/).

Author: I'm not sure it really makes sense to put both links. If you go to one, the other is clearly listed. I'd put "the Python Package Index" in the manuscript only not abstract.

Reviewer: I agree that you could use just one URL.
